front/lib/api/assistant/citations.ts

```diff
@@ -51,50 +52,62 @@ export function citationMetaPrompt() {
 export const getCitationsFromActions = (
   actions: MCPActionType[]
 ): Record<string, CitationType> => {
-  // MCP actions with search results.
   const searchResultsWithDocs = removeNulls(
     actions.flatMap((action) =>
       action.output?.filter(isSearchResultResourceType).map((o) => o.resource)
     )
   );
-  const allMCPSearchResultsReferences = searchResultsWithDocs.reduce<{
-    [key: string]: CitationType;
-  }>(
-    (acc, d) => ({
-      ...acc,
-      [d.ref]: {
-        href: d.uri,
-        title: d.text,
-        provider: d.source.provider ?? "document",
-      },
-    }),
-    {}
-  );
+
+  const searchRefsWithChunks: Record<string, CitationType> = {};
+  searchResultsWithDocs.forEach((d) => {
+    searchRefsWithChunks[d.ref] = {
+      href: d.uri,
+      title: d.text,
+      provider: d.source.provider ?? "document",
+      chunks: d.chunks ?? [],
```

Comment:
I added chunks on the sub agent results and I also added them for search and websearch. I'm not sure this is something we want or not. I'm making some tests locally (that's why the PR is still draft). 
WDYT?

Reply:
I'm not sure I understand, can you give more tokens about what this is doing ?

Reply:
Nevermind it works well without!